src/core_atmosphere/mpas_atm_core_interface.F

```diff
@@ -333,19 +333,178 @@ end function atm_setup_decompositions
    function atm_setup_block(block) result(ierr)
 
       use mpas_derived_types, only : block_type
+      use mpas_pool_routines, only : mpas_pool_get_config
+      use mpas_log, only : mpas_log_write
 
       implicit none
 
       type (block_type), pointer :: block
       integer :: ierr
 
+      integer, pointer :: cam_pcnst
+      integer :: err_level
+
       ierr = 0
 
       call atm_generate_structs(block, block % structs, block % dimensions, block % packages)
 
+      !
+      ! When MPAS-A is operating as a dycore in CAM, the scalars/scalars_tend var_arrays are
+      ! allocated by the call to atm_allocate_scalars, below. The CAM-MPAS interface layer
+      ! should have added a config, cam_pcnst, to the configs pool to indicate how many scalars
+      ! are to be allocated.
+      !
+      if (associated(block % domain % configs)) then
```

Comment:
Whats the reason for this `if` test here? 

My question may come from my unfamiliarity with when `atm_setup_block` is called. In standalone MPAS, is `atm_setup_block` called before `block % domain % configs` is allocated/created?

Reply:
Having looked through the call trees in both CAM-MPAS and stand-alone MPAS-Atmosphere, I think we don't actually need the if-test. Here's the relevant call tree in CAM-MPAS:
```
cam_init
  read_namelist
    dyn_readnl
      cam_mpas_init_phase1
        mpas_allocate_domain
  dyn_grid_init
    cam_mpas_init_phase3
      mpas_bootstrap_framework_phase1
        mpas_block_creator_finalize_block_phase1
          setup_block
```
and here the call tree in stand-alone MPAS-Atmosphere:
```
mpas_init
  mpas_allocate_domain
  setup_namelists
  mpas_bootstrap_framework_phase1
    mpas_block_creator_finalize_block_phase1
      setup_block
```
In both cases, we allocate the domain and setup namelist configs before calling `setup_block`.

I'll do some testing and update this PR with a commit to remove the if-test.